tests/unit/s2n_tls13_cert_verify_test.c

```diff
@@ -270,61 +319,135 @@ int run_tests(const struct s2n_tls13_cert_verify_test *test_case, s2n_mode verif
         struct s2n_signature_scheme test_scheme = *verifying_conn->handshake_params.server_cert_sig_scheme;
         verifying_conn->handshake_params.server_cert_sig_scheme = &test_scheme;
+        verifying_conn->handshake_params.client_cert_sig_scheme = &test_scheme;
+
+        /* Use a hash algorithm different from sender */
         test_scheme.hash_alg = S2N_HASH_SHA1;
-        EXPECT_FAILURE(s2n_tls13_cert_read_and_verify_signature(verifying_conn,
-                verifying_conn->handshake_params.server_cert_sig_scheme));
 
-        /* send and receive with mismatched signature algs */
-        verifying_conn->handshake_params.client_cert_sig_scheme = &test_scheme;
-        test_scheme.hash_alg = S2N_HASH_SHA256;
-        test_scheme.sig_alg = S2N_SIGNATURE_ECDSA;
-        test_scheme.iana_value = 0xFFFF;
+        /* Skip the first two bytes (signature scheme) because s2n_tls13_cert_read_and_verify_signature 
+         * expects first two bytes of the message will be the signature size
+         */
+        EXPECT_SUCCESS(s2n_stuffer_skip_read(&verifying_conn->handshake.io, 2));
 
-        EXPECT_SUCCESS(s2n_hash_init(&verifying_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
-        EXPECT_SUCCESS(s2n_hash_update(&verifying_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
+        EXPECT_FAILURE_WITH_ERRNO(s2n_tls13_cert_read_and_verify_signature(verifying_conn,
+                                          verifying_conn->handshake_params.server_cert_sig_scheme),
+                S2N_ERR_VERIFY_SIGNATURE);
+    };
 
-        EXPECT_SUCCESS(s2n_tls13_cert_verify_send(verifying_conn));
+    /* Verifying connection errors with wrong signature algorithms 
+     * Skip this test when FIPS mode is enabled, since it uses EVP signing method which ignores
+     * signature algorithm mismatch
+     */
+    if (!(s2n_is_in_fips_mode() && s2n_libcrypto_is_awslc())) {
+        /* Derive private/public keys and set connection variables */
+        DEFER_CLEANUP(struct s2n_stuffer certificate_in = { 0 }, s2n_stuffer_free);
+        DEFER_CLEANUP(struct s2n_stuffer certificate_out = { 0 }, s2n_stuffer_free);
+        struct s2n_blob b = { 0 };
+        s2n_pkey_type pkey_type = { 0 };
 
-        EXPECT_SUCCESS(s2n_hash_init(&verifying_conn->handshake.hashes->sha256, S2N_HASH_SHA256));
-        EXPECT_SUCCESS(s2n_hash_update(&verifying_conn->handshake.hashes->sha256, hello, strlen((char *) hello)));
+        DEFER_CLEANUP(struct s2n_connection *verifying_conn = s2n_connection_new(verifier_mode),
+                s2n_connection_ptr_free);
+        DEFER_CLEANUP(struct s2n_connection *sending_conn =
+                              s2n_connection_new(verifier_mode == S2N_CLIENT ? S2N_SERVER : S2N_CLIENT),
+                s2n_connection_ptr_free);
+        EXPECT_NOT_NULL(verifying_conn);
+        EXPECT_NOT_NULL(sending_conn);
+        verifying_conn->actual_protocol_version = S2N_TLS13;
```

Comment:
Nit: Can you move this line to the place where you're configuring the verifying_conn? Same with the all of these test cases. It's good to keep configuration all in the same place.